### PR TITLE
Added support to set placeholder insets

### PR DESCRIPTION
**Pull request by portellaa**
## Pull request checklist

- [x] All tests pass. Demo project builds and runs.
- [x] I have resolved any merge conflicts.
- [x] I have followed the [coding style](https://github.com/jessesquires/HowToContribute#style-guidelines), and reviewed the [contributing guidelines](https://github.com/jessesquires/JSQMessagesViewController/blob/develop/.github/CONTRIBUTING.md). Confirmation: 💪😎👊

#### This fixes issue #
#1907

## What's in this pull request?

Basically the changes are on the `JSQMessagesComposerTextView` class.

### Header File:
- A new property has created in the header file to allow change insets of the placeholder text

### ImplementationFile:
- Set property to default values of (5.0, 7.0, 5.0, 7.0) `UIEdgeInsetsMake(5.0, 7.0, 5.0, 7.0)` which is the same as `CGRectInset(rect, 5.0f, 7.0f)`
- Setter for the property overridden to force layout update when the insets change
- Changed `drawRect:` to use UIEdgeInsetsInsetRect instead of `CGRectInset`

If the developer using the lib doesn't provides any insets, the default values are used.

### Tests
- Changed `textView` init tests to validate if `placeholderInsets` matches the default values
- Added test to check if `placeholderInsets` setter is working correctly

### Notes

Why didn't i use `f` modifier in the `CGFloat` values?
CGFloat is a arch dependent type, which means it is a float in 32bits arch and a double in 64bits arch.
When we use `CGFloat` we force the value to be a float, to let the compiler decide.
I can change it to use `f` if necessary.

Cheers.
Thanks for the awesome work on the lib.

### CODE REVIEW

**Comment by Lucashuang0802** [comment]
Thanks @portellaa for submitting a PR. You might want to change the base branch against `develop` 😃

**Comment by portellaa** [comment]
Hey @Lucashuang0802 changed 😄

File: JSQMessagesViewController/Views/JSQMessagesComposerTextView.h [hunk 1]
```diff
@@ -55,6 +55,12 @@ NS_ASSUME_NONNULL_BEGIN
 @property (strong, nonatomic) UIColor *placeHolderTextColor;
 
 /**
+ *  The insets to be used when the placeholder is drawn. The default value is `UIEdgeInsets(5.0, 7.0, 5.0, 7.0)`.
+ *  Internally the lib uses `UIEdgeInsetsInsetRect` to inset the rect from the `drawRect:` method.
```
Comment (jessesquires) [inline JSQMessagesViewController/Views/JSQMessagesComposerTextView.h]: let's remove this second line

File: JSQMessagesViewController/Views/JSQMessagesComposerTextView.m [hunk 2]
```diff
@@ -169,6 +170,16 @@ - (void)setPlaceHolderTextColor:(UIColor *)placeHolderTextColor
...
 
+- (void)setPlaceHolderInsets:(UIEdgeInsets)placeHolderInsets
+{
+    if (UIEdgeInsetsEqualToEdgeInsets(placeHolderInsets, _placeHolderInsets)) {
+        return;
+    }
+    
+    _placeHolderInsets = placeHolderInsets;
+    [self setNeedsDisplay];
+}
```
Comment (jessesquires) [inline JSQMessagesViewController/Views/JSQMessagesComposerTextView.m]: 💯

File: JSQMessagesViewController/Views/JSQMessagesComposerTextView.m [hunk 3]
```diff
@@ -220,8 +231,8 @@ - (void)drawRect:(CGRect)rect
 
     if ([self.text length] == 0 && self.placeHolder) {
         [self.placeHolderTextColor set];
-
-        [self.placeHolder drawInRect:CGRectInset(rect, 7.0f, 5.0f)
+        
+        [self.placeHolder drawInRect:UIEdgeInsetsInsetRect(rect, self.placeHolderInsets)
```
Comment (jessesquires) [inline JSQMessagesViewController/Views/JSQMessagesComposerTextView.m]: 👍

File: JSQMessagesViewController/Views/JSQMessagesComposerTextView.h (hunk 1 above)
Comment (portellaa) [inline JSQMessagesViewController/Views/JSQMessagesComposerTextView.h]: done 👍

**Comment by portellaa** [comment]
hey @jessesquires 

I changed the documentation to remove the line, but i don't understand the other comments.
Am i missing something?

Cheers 🍻

**Comment by jessesquires** [comment]
Looks good.

Thanks again @portellaa !

**Pull request merged**